daemon.go

```diff
@@ -133,9 +154,9 @@ func (d *daemon) runCheck(writer http.ResponseWriter, uristr string) error {
 		}
 	}
 
-	testHost, err := libp2p.New(libp2p.ConnectionGater(&privateAddrFilterConnectionGater{}))
+	testHost, err := d.createTestHost()
```

Comment:
@aschmahmann Why do we create a new libp2p host for testing?

Reply:
> There's a case here where the host created in this scope successfully connects to one of the maddrs of the passed peerID (using the results from peerAddrsInDHT), but then checkBitswapCID fails with the maddr that was passed in.

This might not be handled as cleanly as it could (particularly in handling things like if the users only wants to pass a `/p2p/<peerID>` multiaddr), but that failure is IMO a good one to have. We want separate checks for:
1. Is the CID (really multihash) advertised in the DHT (or later IPNI)?
2. Are the peer's addresses discoverable (particularly useful if the announcements are DHT based, but also independently useful)
3. Is the peer contactable with the address the user gave us?
4. Is the address the user gave us present in the DHT?
5. Does the peer say they have at least that one CID (doesn't say anything about the rest of any associated DAG but it a quick test and start) when asked using a protocol most clients speak (i.e. currently Bitswap, later also trustless gateways, etc.)?

If we allowed the test to succeed instead of fail we wouldn't have a check on just the passed multiaddr working. We can do something to tease out each test better, but maybe better to just follow that up in another PR

Reply:
Yeah I agree It's very useful to have all those 5 separate checks.

I think we just need to make sure we structure the results better. Will follow up in a separate PR